api/landing.types.ts

```diff
@@ -0,0 +1,42 @@
+export type Testimonial = {
+  title: string;
+  text: string;
+  userName: string;
+  userRole: string;
+  userCompany: string;
+  userImage: {
+    src: string;
+    width: number;
+    height: number;
+  };
+};
+
+export type Company = {
+  name: string;
+  logo: string;
+};
+
+export type Section = {
+  sectionType: string;
+  theme: string;
+  title: string;
+  subtitle?: string;
+  image?: {
+    src: string;
+    width: number;
+    height: number;
+  };
+  companies?: Company[];
+  testimonials?: Testimonial[];
+};
```

Comment:
This is actually 3 different section types, right? There's an alternative to this approach of mixing the types that also requires you to use optional fields:

Union types. So basically you define three separate types.

```
type HeroSection = {
sectionType: 'hero-section'; // or whatever the type was :)
  theme: string;
  title: string;
  subtitle: string;
  image: {
    src: string;
    width: number;
    height: number;
  };
}

type TestimonialSection = {
  testimonials?: Testimonial[];
}

...
```

Then you create a union type from these

```
type Section = HeroSection | TestimonialSection | ...
```

Again, I recommend watching the video solution as this is the approach I'm using. Then you can create a Section component that renders the appropriate section based on the type.